drools-ansible-rulebook-integration-api/src/main/java/org/drools/ansible/rulebook/integration/api/RulesExecutor.java

```diff
@@ -74,6 +74,7 @@ public CompletableFuture<List<Match>> processFacts(String json) {
 
     public CompletableFuture<List<Match>> processEvents(String json) {
         MemoryMonitorUtil.checkMemoryOccupation();
+        rulesEvaluator.validateRulesSetEventStructureOnce(json);
```

Comment:
Shouldn't this validation be performed only if the event doesn't match any rule? If so I believe that this check should be moved after the `processEvents` invocation.

Reply:
@mariofusco Ah, yes. Madhu said "only once when the first event comes thru and none of the rules match". I'll fix.

Reply:
Conditions to execute the validation gets a little tricky. I introduced `RulesSetEventStructure.state`.